bluepysnap/circuit_ids.py

```diff
@@ -0,0 +1,216 @@
+            population_ids = utils.ensure_list(population_ids)
+
+        if isinstance(populations, six.string_types):
+            populations = utils.ensure_list(populations)
+
+        populations = _reformat(populations, population_ids)
+        population_ids = _reformat(population_ids, populations)
+
+        if len(populations) != len(population_ids):
+            raise BluepySnapError("populations and population_ids must have the same size or "
+                                  "being a single value. {} != {}".format(len(populations),
+                                                                          len(population_ids)))
+
+        index = pd.MultiIndex.from_arrays([populations, population_ids])
+        return cls(index, sort_index=sort_index)
+
+    def _locate(self, population):
+        """Returns the index indices corresponding to a given population.
+
+        Args:
+            population (str): the population name you want to locate inside the MultiIndex.
+
+        Returns:
+            numpy.array: indices corresponding to the population.
+        """
+        try:
+            return self.index.get_locs(utils.ensure_list(population))
+        except KeyError:
+            return []
+
+    def filter_population(self, population, inplace=False):
+        """Filter the IDs corresponding to a population.
+
+        Args:
+            population (str): the population you want to extract.
+            inplace (bool): if set to True. Do the transformation inplace.
+
+        Returns:
+            CircuitNodeIds : a filtered CircuitNodeIds containing only IDs for the given population.
+        """
+        if not inplace:
+            return CircuitNodeIds(self.index[self._locate(population)], sort_index=False)
+        self.index = self.index[self._locate(population)]
+        return None
+
+    def get_populations(self, unique=False):
+        """Returns all population values from the circuit node IDs."""
+        if unique:
+            return self.index.levels[0].to_numpy()
+        return self.index.get_level_values(0).to_numpy()
+
+    def get_ids(self, unique=False):
+        """Returns all the ID values from the circuit node IDs."""
+        if unique:
+            return self.index.levels[1].to_numpy()
+        return self.index.get_level_values(1).to_numpy()
+
+    def copy(self):
+        """Copy a CircuitNodeIds."""
+        return CircuitNodeIds(self.index.copy())
+
+    def sort(self, inplace=False):
+        """Sort the CircuitNodeIds by population and then by ids.
+
+        Args:
+            inplace (bool): if set to True. Do the transformation inplace.
+
+        Notes:
+            sorting a CircuitNodeIds will gives better perf for the population extraction and the
+            data extraction from a sorted dataframe.
+        """
+        if not inplace:
+            return CircuitNodeIds(self.index, sort_index=True)
+        self.index = self.index.sortlevel()[0]
+        return None
```

Comment:
It principle : no. But lint was not happy and they recommend to always return a value and in case of inplace like this to return None. 
So, I just followed the recommendations :
https://pycodequ.al/docs/pylint-messages/r1710-inconsistent-return-statements.html